worlds/ahit/Rules.py

```diff
@@ -0,0 +1,988 @@
+           and state.has("Yellow Overpass Manhole Cleared", world.player) \
+           and state.has("Green Clean Station Cleared", world.player) \
+           and state.has("Green Clean Manhole Cleared", world.player) \
+           and state.has("Bluefin Tunnel Cleared", world.player) \
+           and state.has("Pink Paw Station Cleared", world.player) \
+           and state.has("Pink Paw Manhole Cleared", world.player)
+
+
+def set_rules(world: "HatInTimeWorld"):
+    # First, chapter access
+    starting_chapter = ChapterIndex(world.options.StartingChapter.value)
+    world.set_chapter_cost(starting_chapter, 0)
+
+    # Chapter costs increase progressively. Randomly decide the chapter order, except for Finale
+    chapter_list: List[ChapterIndex] = [ChapterIndex.MAFIA, ChapterIndex.BIRDS,
+                                        ChapterIndex.SUBCON, ChapterIndex.ALPINE]
+
+    final_chapter = ChapterIndex.FINALE
+    if world.options.EndGoal.value == 2:
+        final_chapter = ChapterIndex.METRO
+        chapter_list.append(ChapterIndex.FINALE)
+    elif world.options.EndGoal.value == 3:
+        final_chapter = None
+        chapter_list.append(ChapterIndex.FINALE)
+
+    if world.is_dlc1():
+        chapter_list.append(ChapterIndex.CRUISE)
+
+    if world.is_dlc2() and final_chapter is not ChapterIndex.METRO:
+        chapter_list.append(ChapterIndex.METRO)
+
+    chapter_list.remove(starting_chapter)
+    world.random.shuffle(chapter_list)
+
+    if starting_chapter is not ChapterIndex.ALPINE and (world.is_dlc1() or world.is_dlc2()):
+        index1: int = 69
+        index2: int = 69
+        pos: int
+        lowest_index: int
+        chapter_list.remove(ChapterIndex.ALPINE)
+
+        if world.is_dlc1():
+            index1 = chapter_list.index(ChapterIndex.CRUISE)
+
+        if world.is_dlc2() and final_chapter is not ChapterIndex.METRO:
+            index2 = chapter_list.index(ChapterIndex.METRO)
+
+        lowest_index = min(index1, index2)
+        if lowest_index == 0:
+            pos = 0
+        else:
+            pos = world.random.randint(0, lowest_index)
+
+        chapter_list.insert(pos, ChapterIndex.ALPINE)
+
+    if world.is_dlc1() and world.is_dlc2() and final_chapter is not ChapterIndex.METRO:
+        chapter_list.remove(ChapterIndex.METRO)
+        index = chapter_list.index(ChapterIndex.CRUISE)
+        if index >= len(chapter_list):
+            chapter_list.append(ChapterIndex.METRO)
+        else:
+            chapter_list.insert(world.random.randint(index+1, len(chapter_list)), ChapterIndex.METRO)
+
+    lowest_cost: int = world.options.LowestChapterCost.value
+    highest_cost: int = world.options.HighestChapterCost.value
+    cost_increment: int = world.options.ChapterCostIncrement.value
+    min_difference: int = world.options.ChapterCostMinDifference.value
+    last_cost = 0
+    cost = 0
+    loop_count = 0
+
+    for chapter in chapter_list:
```

Comment:
Could use `enumerate()` here instead of manually tracking `loop_count`

Reply:
done